cmake/projects/OpenSSL/schemes/url_sha1_openssl.cmake.in

```diff
@@ -50,11 +52,17 @@ if(ANDROID)
     hunter_user_error("Could not find android architecture. Please set ANDROID_ARCH_NAME in your toolchain or use CMake 3.7+")
   endif()
   set(configure_command
-      # Ignored. Prevents script from checking host uname
-      RELEASE=2.6.37
-      SYSTEM=android
-      ARCH=${ANDROID_SSL_ARCH}
-      ${configure_command})
+      ${configure_command} ${CMAKE_C_FLAGS})
+
+  # Latest OpenSSL (1.1.1) still does not have correct android targets, so set generic with our CFLAGS
+  string(COMPARE EQUAL "${ANDROID_SSL_ARCH}" "arm64" _is_arm64)
+  string(COMPARE EQUAL "${ANDROID_SSL_ARCH}" "mips64" _is_mips64)
+  string(COMPARE EQUAL "${ANDROID_SSL_ARCH}" "x86_64" _is_x8664)
+  if (_is_arm64 OR _is_mips64 OR _is_x8664)
+    set(configure_opts "linux-generic64")
```

Comment:
> It only looks like a big change because of the way to check 64-bitness

It doesn't look so for me. Before it was `RELEASE=.. SYSTEM=android ARCH=... ./config` and now it's `./Configure linux-generic*`. Please add the link with something like "If you want to build OpenSSL for Android do ...". Currently you're using `./Configure` for Android build but the link goes to `./config` script - seems not very consistent for me.